nodes/src/nodes/memory_persistent/memory_store.py

```
@@ -0,0 +1,710 @@
+    def replace_history(self, session_id: str, entries: List[Dict[str, Any]]) -> None:
+        """Replace all history entries for a session with the given list."""
+
+    def increment(self, session_id: str, key: str, amount: int = 1) -> Dict[str, Any]:
+        """Atomically increment a numeric value. Default uses get+put; backends may override."""
+        result = self.get(session_id, key)
+        current = result.get('value', 0) if result.get('ok') else 0
+        new_value = current + amount
+        return self.put(session_id, key, new_value)
+
+    def close(self) -> None:
+        """Release any resources held by the backend. No-op by default."""
```

Ruff (0.15.9)

[warning] `MemoryBackend.close` is an empty method in an abstract base class, but has no abstract decorator

(B027)